driver-core/src/main/com/mongodb/internal/connection/OidcAuthenticator.java

```diff
@@ -282,10 +287,12 @@ private void authenticationLoop(final InternalConnection connection, final Conne
     }
 
     private void authenticationLoopAsync(final InternalConnection connection, final ConnectionDescription description,
+            final OperationContext operationContext,
             final SingleResultCallback<Void> callback) {
         fallbackState = FallbackState.INITIAL;
         beginAsync().thenRunRetryingWhile(
-                c -> super.authenticateAsync(connection, description, c),
+                operationContext.getTimeoutContext(),
```

Comment:
(See prior comment).

I believe that no timeout context should be passed here, and that in general `thenRunRetryingWhile` should not take a timoutContext as a parameter.